operator-framework-core/src/main/java/io/javaoperatorsdk/operator/api/config/ConfigurationService.java

```diff
@@ -448,6 +452,10 @@ default boolean previousAnnotationForDependentResourcesEventFiltering() {
     return true;
   }
 
+  default List<Class<? extends HasMetadata>> previousAnnotationUsageBlacklist() {
+    return List.of(Deployment.class, StatefulSet.class, ReplicaSet.class);
```

Comment:
@shawkins do I remember correctly these are the ones you had issues with? thx!

Reply:
Deployment and StatefulSet I believe are the ones we encounted problems with before. DaemonSet may need to be included as well.

Reply:
thank you!

Reply:
just for the record a I did not add DaemonSet since was not able to reproduce the behavior regarding generation increase with it.